src/extensions/nginx-app-protect/monitoring/processor/nap.go

```diff
@@ -506,13 +505,13 @@ func setValue(napConfig *NAPConfig, key, value string, logger *logrus.Entry) err
 	case httpServerPort:
 		napConfig.HTTPServerPort = value
 	case httpURI:
-		napConfig.HTTPURI = value
+		napConfig.HTTPURI = strings.ReplaceAll(value, encodedComma, decodedComma)
 	case isTruncated:
 		napConfig.IsTruncated = value
 	case policyName:
 		napConfig.PolicyName = value
 	case request:
-		napConfig.Request = value
+		napConfig.Request = strings.ReplaceAll(value, encodedComma, decodedComma)
```

Comment:
I have a question. So are the `HTTPURI` and `Request` fields fully URL encoded or just the commas? If they are fully encoded, should we be decoding them? The implementation we have now only decodes commas.
```go
request, err := url.QueryUnescape(value)
```

Reply:
only the comma is encoded in `httpURI` and `request`